src/utils/utils_posix_common.c

```diff
@@ -266,3 +266,27 @@ int utils_file_open_or_create(const char *path) {
 
     return fd;
 }
+
+// Expected input:
+// char *str_threshold = utils_env_var_get_str("UMF_PROXY", "size.threshold=");
+size_t utils_get_size_threshold(char *str_threshold) {
+    if (!str_threshold) {
+        return 0;
+    }
+
+    // move to the beginning of the number
+    str_threshold += strlen("size.threshold=");
+
+    long long_threshold = atol(str_threshold);
+    if (long_threshold < 0) {
+        LOG_ERR("size threshold cannot be negative! = (char *) %s, (int) %li",
+                str_threshold, long_threshold);
+        return 0;
+    }
+
+    size_t int_threshold = (size_t)long_threshold;
```

Comment:
nit: `int_threshold` (along with the `(int)` part in the log below) seems like a weird name here, now, perhaps just `threshold` (and `size_t` in the log below)?